client/api_object.go

```diff
@@ -730,36 +730,76 @@ func GetSegmentEnd(begin int64, total int64, per int64) int64 {
 	return begin + per - 1
 }
 
-// FGetObjectResumable download s3 object payload and return the related object info
+// FGetObjectResumable download s3 object payload with resumable download
 func (c *client) FGetObjectResumable(ctx context.Context, bucketName, objectName, filePath string, opts types.GetObjectOptions) error {
-	tempFilePath := filePath + types.TempFileSuffix
+	// Get the object detailed meta for object whole size
+	meta, err := c.HeadObject(ctx, bucketName, objectName)
+	if err != nil {
+		return err
+	}
+
+	tempFilePath := filePath + "_" + c.defaultAccount.GetAddress().String() + opts.Range + types.TempFileSuffix
 
 	var (
-		startOffset  int64
-		segmentSize  int64
-		objectOption types.GetObjectOptions
-		segNum       int64
+		startOffset    int64
+		endOffset      int64
+		partEndOffset  int64
+		maxSegmentSize int64
+		objectOption   types.GetObjectOptions
+		segNum         int64
+		partSize       int64
 	)
 
+	// 1) check paramter
 	params, err := c.GetParams()
 	if err != nil {
 		return err
 	}
-	segmentSize = int64(params.GetMaxSegmentSize())
+	maxSegmentSize = int64(params.GetMaxSegmentSize())
+
+	// minPartSize: 16MB
+	if opts.PartSize == 0 {
+		opts.PartSize = types.MinPartSize
+	}
+
+	partSize = int64(opts.PartSize)
+	if partSize%maxSegmentSize != 0 {
+		return errors.New("part size should be an integer multiple of the segment size")
+	}
 
+	isRange, rangeStart, rangeEnd := utils.ParseRange(opts.Range)
```

Comment:
If first download finish half object,  and second download with different range info , how to handle this solution?

Reply:
You may add comments in  order force the user pass the same range params